### DISCUSSION TITLE
Next.js production mode conflicts with Python app on same port

### DISCUSSION THREAD

**Discussion by hmaesta** [vercel/next.js#60497] [Help]
### Summary

Running a Python app and a Next.js app on port 3000 works fine in Next.js dev mode (`npm run dev`), but in production mode (`npm run build && npm run start`) the Next.js app seems to take control of all routes, causing 404 errors for the Python app's routes. 

Why that happens? 
Is it possible to resolve this routing issue?

#### Additional information

- Next.js 12
- React 18.2
- Node 20.9

**Comment by don-esteban** [comment]
Please describe your use case. Never heard of binding two apps to one port. I only heard of "combining" different apps and services with proxys/rewrites.

**Reply by hmaesta** [reply]
In production it runs under proxies and everything is fine — but I was wondering if it's possible to get the same `npm run dev` behavior using `npm run start` to improve the accuracy of the front-end tests. 

When running `npm run start` everything looks fine except the back-end (Python app) failing and it got me curious about what is different between those environments configurations and if there's a way to get around it.
